### PR TITLE
Update LUIS authoring SDK to reflect latest changes

### PR DESCRIPTION
**Pull request by AmirGeorge**
## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [ ] Please add REST spec PR link to the SDK PR
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

File: src/SDKs/CognitiveServices/dataPlane/Language/LUIS/Authoring/Generated/Models/ApplicationPublishObject.cs [hunk 1]
```diff
@@ -32,13 +32,10 @@ public ApplicationPublishObject()
...
         /// <param name="isStaging">Indicates if the staging slot should be
         /// used, instead of the Production one.</param>
-        /// <param name="region">The target region that the application is
-        /// published to.</param>
-        public ApplicationPublishObject(string versionId = default(string), bool? isStaging = default(bool?), string region = default(string))
+        public ApplicationPublishObject(string versionId = default(string), bool? isStaging = default(bool?))
         {
             VersionId = versionId;
             IsStaging = isStaging;
-            Region = region;
```
Comment (yangyuan) [inline src/SDKs/CognitiveServices/dataPlane/Language/LUIS/Authoring/Generated/Models/ApplicationPublishObject.cs]: This (I guess) should be considered as a breaking change. Thus might require a major version bump.

**Review by yangyuan — APPROVED** [review]

**Comment by dsgouda** [comment]
@AmirGeorge please fix failing tests

**Comment by AmirGeorge** [comment]
> @AmirGeorge please fix failing tests

@dsgouda any input on why the delete app API is now failing? https://azuresdkci.westus2.cloudapp.azure.com/job/CI-NetSDK_U/1065/console

The only change added to it was to add an optional boolean query parameter.

**Comment by dsgouda** [comment]
@AmirGeorge A couple of things to check
- Please check if the delete operation URL was modified in the spec
- If not, you may have to re-record your tests

Also, do these tests work on your local machine?

**Comment by AmirGeorge** [comment]
> @AmirGeorge A couple of things to check
> 
> * Please check if the delete operation URL was modified in the spec
> * If not, you may have to re-record your tests
> 
> Also, do these tests work on your local machine?

@dsgouda No they do not work on my local machine. The test project was even build failing locally and I had to add xunit reference in the project file (added in this PR) to get it to build. Now when I try to run the tests from VS test explorer, it instantly says 'Test run finished' without actually running anything.

**Comment by dsgouda** [comment]
@AmirGeorge could you add the link to your REST spec PR and check the first bullet point

**Comment by AmirGeorge** [comment]
> @AmirGeorge could you add the link to your REST spec PR and check the first bullet point

@dsgouda https://github.com/Azure/azure-rest-api-specs/commit/7d344e289c4befb3cf2d92baee7ea1af204faae8

The only change is the addition of the optional force flag (topmost change in the linked commit). 

Could the query param be somehow interpreted as part of the url path not as a query param? The error message suggests this:
Unable to find a matching HTTP request for URL 'DELETE /luis/api/v2.0/apps/0645d086-d6a3-4a4b-9a9d-5efaab7bbd71?force=false'.

**Comment by dsgouda** [comment]
@AmirGeorge that's right,
The bool is a Path parameter as indicated [here](https://github.com/Azure/azure-rest-api-specs/commit/7d344e289c4befb3cf2d92baee7ea1af204faae8#diff-b30ca75196a51d0afe8d4bcf3d5f48d1R9368)
You may have to re-record the tests

**Comment by AmirGeorge** [comment]
> @AmirGeorge that's right,
> The bool is a Path parameter as indicated [here](https://github.com/Azure/azure-rest-api-specs/commit/7d344e289c4befb3cf2d92baee7ea1af204faae8#diff-b30ca75196a51d0afe8d4bcf3d5f48d1R9368)
> You may have to re-record the tests

@dsgouda I am not sure how to do that. Any pointers?

**Comment by dsgouda** [comment]
@AmirGeorge instructions to record tests can be found [here](https://github.com/Azure/azure-sdk-for-net/blob/master/Documentation/Using-Azure-TestFramework.md)

**Comment by dsgouda** [comment]
@AmirGeorge Please check if the tests work for you in PlayBack mode. Also, please ensure you have copied over the newly recorded test jsons to the SessionRecords directory

**Comment by AmirGeorge** [comment]
@dsgouda finally done.

FYI the link to record tests was a bit confusing. It was not clear whether I should start from creating a service principal or what. I ended up just changing the `Playback` mode in `BaseTest` to `Record` and assign a valid subscription key.

It took a lot of time because at first the tests project was failing to build locally, so I added the xunit nuget package:
`<PackageReference Include="xunit" Version="2.4.1" />`

After this it could build, but on trying to run the tests from tests explorer, the test run exists instantly without actually running anything. Doing some research, I arrived that I need to add the following 2 references:
`<PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.9.0" />`
`<PackageReference Include="xunit.runner.visualstudio" Version="2.4.1">`

After that the tests worked in playback mode. I changed to record and re-recorded all the tests with many fixes to get them working.

**Comment by dsgouda** [comment]
> 
> 
> @dsgouda finally done.
> 
> FYI the link to record tests was a bit confusing. It was not clear whether I should start from creating a service principal or what. I ended up just changing the `Playback` mode in `BaseTest` to `Record` and assign a valid subscription key.
> 
> It took a lot of time because at first the tests project was failing to build locally, so I added the xunit nuget package:
> `<PackageReference Include="xunit" Version="2.4.1" />`
> 
> After this it could build, but on trying to run the tests from tests explorer, the test run exists instantly without actually running anything. Doing some research, I arrived that I need to add the following 2 references:
> `<PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.9.0" />`
> `<PackageReference Include="xunit.runner.visualstudio" Version="2.4.1">`
> 
> After that the tests worked in playback mode. I changed to record and re-recorded all the tests with many fixes to get them working.

That's interesting, @shahabhijeet has a better handle on test dependencies and can investigate the issue you faced. For now, I can review rest of the PR since the checks are passing.
Thanks for the updates

File: src/SDKs/CognitiveServices/dataPlane/Language/LUIS/Authoring.Tests/Microsoft.Azure.CognitiveServices.LUIS.Authoring.Tests.csproj [hunk 2]
```diff
@@ -9,6 +9,12 @@
 
   <ItemGroup>
     <PackageReference Include="Microsoft.Azure.Management.ResourceManager" Version="1.6.0-preview" />
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.9.0" />
+    <PackageReference Include="xunit" Version="2.4.1" />
+    <PackageReference Include="xunit.runner.visualstudio" Version="2.4.1">
+      <PrivateAssets>all</PrivateAssets>
```
Comment (dsgouda) [inline src/SDKs/CognitiveServices/dataPlane/Language/LUIS/Authoring.Tests/Microsoft.Azure.CognitiveServices.LUIS.Authoring.Tests.csproj]: Do you need to add the `PrivateAssets` and `IncludeAssets` tags?

**Review by dsgouda — CHANGES_REQUESTED** [review]
Looks good for the most part

File: src/SDKs/CognitiveServices/dataPlane/Language/LUIS/Authoring.Tests/Microsoft.Azure.CognitiveServices.LUIS.Authoring.Tests.csproj (hunk 2 above)
Comment (AmirGeorge) [reply src/SDKs/CognitiveServices/dataPlane/Language/LUIS/Authoring.Tests/Microsoft.Azure.CognitiveServices.LUIS.Authoring.Tests.csproj]: @dsgouda done.

**Review by dsgouda — APPROVED** [review]
LGTM

**Pull request merged**